t/api/04-jobs.t

```diff
@@ -1636,7 +1636,7 @@ subtest 'handle git_clone with CASEDIR' => sub {
     my $gru_task_values = shift @gru_args;
     is_deeply $gru_task_values,
       {
-        't/data/openqa/share/tests/foobar' => 'http://localhost/foo.git',
+        path('t/data/openqa/share/tests/foobar')->realpath => 'http://localhost/foo.git',
         't/data/openqa/share/tests/foobar/needles' => 'http://localhost/bar.git'
```

Comment:
If anyone wonders why I don't expand the path for needles,
let me explain that `Mojo::File` behaves a bit odd.
Actually neither of those paths exist. But if at least the path up to the last path element (e.g. `t/data/openqa/share/tests/`) exists, `path()` will not raise an error when being called with an unexisting path like `t/data/openqa/share/tests/foobar`.

```
perl> use Mojo::File qw(path);
perl> path("t/data/openqa/share/tests/foobar")->realpath;p
/home/dheidler/openqa/repos/openQA/t/data/openqa/share/tests/foobar
perl> path("t/data/openqa/share/tests/foobar/needles")->realpath;
Invalid path at (eval 36) line 1.
```

```
$ ls -l t/data/openqa/share/tests/
insgesamt 0
-rw-r--r-- 1 dheidler users  0 30. Okt 2023  archive.tar.xz
drwxr-xr-x 1 dheidler users 14 12. Sep 2022  archlinux
drwxr-xr-x 1 dheidler users 50 30. Sep 11:38 example
drwxr-xr-x 1 dheidler users 14 12. Sep 2022  fedora
drwxr-xr-x 1 dheidler users 56 18. Okt 16:41 opensuse
drwxr-xr-x 1 dheidler users 18 30. Okt 2023  Unicorn